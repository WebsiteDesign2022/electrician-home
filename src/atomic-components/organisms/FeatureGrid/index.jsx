import React from 'react';
import FeatureBox from '../../molecules/FeatureBox';

import './index.scss';

const FeatureGrid = ({ featureList }) => (
    <div className="bg-black">
        <div className="container px-4 py-5" id="hanging-icons">
            <div className="row g-4 py-5 row-cols-1 row-cols-sm-3">
                { featureList.map(({ title, description }, index) => ( <FeatureBox title={title} description={description} key={index}/> )) }
            </div>
        </div>
    </div>
);

export default FeatureGrid;