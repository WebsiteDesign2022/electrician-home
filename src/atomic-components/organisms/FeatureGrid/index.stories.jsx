import React from 'react';

import FeatureGrid from '.';

export default {
  title: 'Electrician-Home/organisms/FeatureGrid',
  component: FeatureGrid,
  parameters: {
    // More on Story layout: https://storybook.js.org/docs/react/configure/story-layout
    layout: 'fullscreen',
  },
};

const Template = (args) => <FeatureGrid {...args} />;

export const Primary = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Primary.args = {
    featureList: [
        {
            title: 'Primary title', 
            description: 'Primary description. Primary description. Primary description.' 
        },
        {
            title: 'Primary title', 
            description: 'Primary description. Primary description.' 
        },
        {
            title: 'Primary title', 
            description: 'Primary description' 
        },
        {
            title: 'Primary title', 
            description: 'Primary description' 
        },
        {
            title: 'Primary title', 
            description: 'Primary description. Primary description. ' 
        },
        {
            title: 'Primary title', 
            description: 'Primary description. Primary description. Primary description.' 
        }
    ]
};
