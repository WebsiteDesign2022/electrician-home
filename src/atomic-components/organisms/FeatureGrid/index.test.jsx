import TestRenderer from 'react-test-renderer';

import FeatureGrid from '.';

describe("FeatureGrid", () => {
    test("renders FeatureGrid given empty", () => {
        const featureGrid = TestRenderer.create(
            <FeatureGrid
                featureList={[]}
            />
        );
        expect(featureGrid.toJSON()).toMatchSnapshot();
    });
    test("renders FeatureGrid given multiple feature items", () => {
        const featureGrid = TestRenderer.create(
            <FeatureGrid
                featureList={[
                    {
                        title: 'feature-title 1',
                        description: 'feature-description 1'
                    },
                    {
                        title: 'feature-title 2',
                        description: 'feature-description 2'
                    },
                ]}
            />
        );
        expect(featureGrid.toJSON()).toMatchSnapshot();
    });
});
