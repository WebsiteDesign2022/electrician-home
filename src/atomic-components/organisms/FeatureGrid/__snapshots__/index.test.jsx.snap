// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureGrid renders FeatureGrid given empty 1`] = `
<div
  className="bg-black"
>
  <div
    className="container px-4 py-5"
    id="hanging-icons"
  >
    <div
      className="row g-4 py-5 row-cols-1 row-cols-sm-3"
    />
  </div>
</div>
`;

exports[`FeatureGrid renders FeatureGrid given multiple feature items 1`] = `
<div
  className="bg-black"
>
  <div
    className="container px-4 py-5"
    id="hanging-icons"
  >
    <div
      className="row g-4 py-5 row-cols-1 row-cols-sm-3"
    >
      <div
        className="col d-flex align-items-start"
      >
        <div
          className="icon-square flex-shrink-0 me-3 py-1 arrow"
        >
          &gt;&gt;
        </div>
        <div
          className="align-items-start"
        >
          <h2
            align="left"
            className="title"
          >
            feature-title 1
          </h2>
          <p
            align="left"
            className="description"
          >
            feature-description 1
          </p>
        </div>
      </div>
      <div
        className="col d-flex align-items-start"
      >
        <div
          className="icon-square flex-shrink-0 me-3 py-1 arrow"
        >
          &gt;&gt;
        </div>
        <div
          className="align-items-start"
        >
          <h2
            align="left"
            className="title"
          >
            feature-title 2
          </h2>
          <p
            align="left"
            className="description"
          >
            feature-description 2
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
